spec/jobs/mass_award_badge_spec.rb

```diff
@@ -35,14 +26,26 @@
 
       UserBadge.create!(badge_id: Badge::Member, user: user, granted_by: Discourse.system_user, granted_at: Time.now)
 
-      execute_job([user.email, user_2.email])
+      execute_job(user)
+      execute_job(user_2)
 
       expect(UserBadge.find_by(user: user, badge: badge).featured_rank).to eq(2)
       expect(UserBadge.find_by(user: user_2, badge: badge).featured_rank).to eq(1)
     end
 
-    def execute_job(emails)
-      subject.execute(users_batch: emails, badge_id:  badge.id, mode: 'email')
+    it 'grants a badge multiple times to a user' do
+      badge.update!(multiple_grant: true)
+      Notification.destroy_all
+      execute_job(user, count: 4, grant_existing_holders: true)
+      instances = UserBadge.where(user: user, badge: badge)
+      expect(instances.count).to eq(4)
+      notifications = Notification.where(user: user)
+      expect(notifications.count).to eq(1)
+      expect(instances.map(&:notification_id).uniq).to eq([notifications.first.id])
```

Comment:
```suggestion
      expect(instances.map(&:notification_id).uniq).to contain_exactly(notifications.first.id)
```

Reply:
I also think we should test for the sequences to make sure the seq has been set correctly.